### PR TITLE
New: Multiput Upload Part 3

### PR DESCRIPTION
**Pull request by wenboyu2**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @wenboyu2 has signed the CLA. Thanks for the pull request!

File: src/api/uploads/BaseUpload.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        const { response } = error;
+        if (!response) {
+            return {};
+        }
+
+        if (response.status === 401 || response.status === 403) {
+            return response;
+        }
+
+        return response.json();
```
Comment (priyajeet) [inline src/api/uploads/BaseUpload.js]: This also fails for 204 response, just FYI

File: src/api/uploads/__tests__/MultiputUpload-test.fixme [hunk 2]
```diff
@@ -15,49 +15,26 @@ const config = {
 };
 let file;
 const createSessionUrl = 'https://test.box.com/createSession';
-const destinationFolderId = '123';
+const partSize = 9;
 
-describe('api/MultiputUpload', () => {
+describe.only('api/MultiputUpload', () => {
```
Comment (priyajeet) [inline src/api/uploads/__tests__/MultiputUpload-test.fixme]: Weren't you planning on putting the new changes here in a new file?

Comment (wenboyu2) [reply src/api/uploads/__tests__/MultiputUpload-test.fixme]: Yes, but these changes are for fixing the old test. If you've already converted/ing these tests, I can also fix the jest tests. I'll still write new tests in a separate file as we've discussed.

File: src/api/uploads/BaseUpload.js (hunk 1 above)
Comment (wenboyu2) [reply src/api/uploads/BaseUpload.js]: yeah but when it's 204 `getErrorResponse` shouldn't get called

File: src/api/uploads/__tests__/MultiputUpload-test.fixme (hunk 2 above)
Comment (wenboyu2) [reply src/api/uploads/__tests__/MultiputUpload-test.fixme]: I made these changes a long time ago, and when I apply the stash git automatically applied them here, didn't bother to extract the changes out

File: src/api/APIFactory.js [hunk 3]
```diff
@@ -46,6 +47,11 @@ class APIFactory {
      */
     chunkedUploadAPI: ChunkedUploadAPI;
 
+    /**
+     * @property {MultiputUploadAPI}
+     */
+    multiputUploadAPI: MultiputUploadAPI;
```
Comment (klk-) [inline src/api/APIFactory.js]: I don't think we use the word 'multiput' in any public documentation.  Also, it is confusing at first glance that we have both a chunked upload API and a multiput API.

File: src/api/uploads/BaseUpload.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+     * @return {Promise}
+     */
+    resolveConflict = async (response: Object): Promise<> => {
+        if (this.overwrite && response.context_info) {
+            this.fileId = response.context_info.conflicts.id;
+            return;
+        }
+
+        const extension = this.fileName.substr(this.fileName.lastIndexOf('.')) || '';
+        this.fileName = `${this.fileName.substr(0, this.fileName.lastIndexOf('.'))}-${Date.now()}${extension}`;
```
Comment (klk-) [inline src/api/uploads/BaseUpload.js]: Add a comment with an example of the file name this line produces.

File: src/api/APIFactory.js (hunk 3 above)
Comment (wenboyu2) [reply src/api/APIFactory.js]: yeah I'll rename everything to chunked in a later PR.
Removing this line and related lines from this file for now.

File: src/api/uploads/BaseUpload.js (hunk 4 above)
Comment (wenboyu2) [reply src/api/uploads/BaseUpload.js]: done

**Pull request merged**